File: lib/src/main/java/arsatech/co/utils/NumberFormatUtils.java
package arsatech.co.utils;

import android.annotation.SuppressLint;

import java.text.DecimalFormat;

/**
 * Created by Aref Bahreini Nejad on 27/04/2019.
 * Updated on 27/04/2019
 */
public class NumberFormatUtils {

	@SuppressLint("DefaultLocale")
	public static String digitDelimiter(long number) {
		// The comma in the format specifier does the trick
		return String.format("%,d", number);
	}

	@SuppressLint("DefaultLocale")
	public static String numberFormatter(double number, int decimalPrecision) {
		return String.format("%,." + decimalPrecision + "f", number);
	}

	@SuppressLint("DefaultLocale")
	public static String decimalFormatter(double number, int decimalPrecision) {
		if (decimalPrecision == 0)
			return String.format("%,." + decimalPrecision + "f", number);
		String precision = String.format("%0" + decimalPrecision + "d", 0);
		DecimalFormat formatter = new DecimalFormat("#,###." + precision);
		return formatter.format(number);
	}

}

File: lib/src/main/kotlin/com/arefdev/base/utils/NumberFormatUtils.kt
package com.arefdev.base.utils

import android.annotation.SuppressLint
import java.text.DecimalFormat

/**
 * Updated on 21/09/2023
 *
 * @author <a href="https://github.com/arefbhrn">Aref Bahreini</a>
 */
object NumberFormatUtils {
    @SuppressLint("DefaultLocale")
    fun digitDelimiter(number: Long): String {
        // The comma in the format specifier does the trick
        return String.format("%,d", number)
    }

    @SuppressLint("DefaultLocale")
    fun numberFormatter(number: Double, decimalPrecision: Int): String {
        return String.format("%,." + decimalPrecision + "f", number)
    }

    @SuppressLint("DefaultLocale")
    fun decimalFormatter(number: Double, decimalPrecision: Int): String {
        if (decimalPrecision == 0)
            return String.format("%,.0f", number)
        val precision = String.format("%0" + decimalPrecision + "d", 0)
        val formatter = DecimalFormat("#,###.$precision")
        return formatter.format(number)
    }
}
